File: src/routing/direct/DirectNode.java
package routing.direct;

import node.Node;

import java.util.UUID;

import java.util.Set;
import java.util.HashSet;

public class DirectNode implements Node {
  private UUID id;
  private Set<Node> connections = new HashSet<Node>();

  public UUID id(){
    return id;
  }

  public void link(Node other) {
    connections.add(other);
  }

  public void unlink(Node other) {
    connections.remove(other);
  }

  public void share(Block block);
}

File: src/main/scala/network/direct/DirectNode.scala
package main.scala.network.direct

import main.scala.node.Node

class DirectNode extends Node[DirectNode] {
  override val connections = scala.collection.mutable.Set[DirectNode]()

  override def link(that: DirectNode) {
    super.link(that)
    connections += that
  }

  override def unlink(that: DirectNode) {
    super.link(that)
    connections -= that
  }
}

